### PR TITLE
Allow custom amount on braintree subscriptions (#546)

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/solitude/issues/546

### CODE REVIEW

File: lib/brains/tests/test_forms.py [hunk 1]
```diff
@@ -51,23 +54,77 @@ def test_parse_ok(self):
...
+        assert 'plan' in form.errors, form.errors.as_text()
+
+    def test_plan_amount_is_not_customizable(self):
+        form = self.submit(expect_errors=True, data={
+            # Link to a non-donation product.
+            'plan': 'moz-brick',
+            # You cannot change the price on a product with a fixed amount.
+            'amount': '1.99',
+        })
+        assert 'amount' in form.errors, form.errors.as_text()
```
Comment (muffinresearch) [inline lib/brains/tests/test_forms.py]: Since there's more than one error for amount could these tests check the code?

**Comment by muffinresearch** [comment]
I think it might be nice to make the error assertions a bit more specific where possible. Just helps ensure the right error is happening at the right place and will also help with the future when more error cases are added under the same key.

r+wc

File: lib/brains/tests/test_forms.py (hunk 1 above)
Comment (kumar303) [inline lib/brains/tests/test_forms.py]: oh yeah, good call. I don't know if I can get at the code but I can pattern match part of the message

Comment (kumar303) [inline lib/brains/tests/test_forms.py]: done

**Pull request merged**